app/src/androidTest/java/org/mozilla/focus/activity/OnBoardingTest.java

```diff
@@ -45,18 +47,38 @@ public void setUp() {
         activityTestRule.launchActivity(new Intent());
     }
 
+    /**
+     * Test case no: TC_0033
+     * Test case name: Go through onboarding
+     * Steps:
+     * 1. Launch Rocket
+     * 2. check turbo mode is selected on first onbording page
+     * 3. move forward and backfard to check 4 pages of onboarding are displayed correctly
+     * 4. open menu
+     * 5. check turbo mode is selected
+     */
+
     @Test
     public void turnOnTurboModeDuringOnBoarding_turboModeIsOnInMenu() {
 
         // Check if turbo mode switch is on
         onView(allOf(withId(R.id.switch_widget), isDisplayed())).check(matches(isChecked()));
 
-        // Click next button in the first on boarding page
+        // Click next button in the first on-boarding page
         onView(allOf(withId(R.id.next), isDisplayed())).perform(click());
 
-        // Click next button in the second on boarding page
+        // Click next button in the second on-boarding page
         onView(allOf(withId(R.id.next), isDisplayed())).perform(click());
 
+        // Swipe right to go to second on-boarding page
+        onView(allOf(withId(R.id.image), isDisplayed())).perform(swipeRight());
+
+        IdlingRegistry.getInstance().register();
```

Comment:
you don't need this line

Reply:
Checked on issue [Espresso can not wait until the swipe action is done](https://stackoverflow.com/questions/37294132/espresso-doesnt-wait-for-swipe-action-on-a-viewpager-to-be-finished) so that I need to implement idling resource to wait for the thread, somehow it works but after your suggestion I consult on
 [idling resource google sample ](https://github.com/googlesamples/android-testing/blob/master/ui/espresso/IdlingResourceSample/app/src/androidTest/java/com/example/android/testing/espresso/IdlingResourceSample/ChangeTextBehaviorTest.java) I think I may need register the target idling resource as a correct way for it.

Reply:
after thoroughly study I add ViewPagerIdlingResource class to apply on OnBoardingTest  so that Espresso will wait until view pager idling to take next action, thanks !